[Renee] iP

reneeyeow02 opened the pull request:
# MeowMeow (=^ↀwↀ^=) 

>Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

MeowMeow will make sure that you never have to remember anything ever again!!! It's 

* text-based
* easy to learn
* ~~FAST~~ _SUPER FAST_ to use

All you need to do is:

1. download it from here.
2. double-click it. 
3. add your tasks. 
4. let it manage your tasks for you 😸

And it's **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

reneeyeow02 committed:
Level 1. Greet, Echo, Exit

reneeyeow02 committed:
Level 2. Add, List

reneeyeow02 committed:
Level 3. Mark as Done

reneeyeow02 committed:
Level 4. ToDos, Events, Deadlines

reneeyeow02 committed:
Duke A-TextUiTesting: Automated Text UI Testing

reneeyeow02 committed:
Level 5. Handle Errors

reneeyeow02 committed:
Level 6. Delete

reneeyeow02 committed:
Update EXPECTED.TXT

reneeyeow02 committed:
Fix indentation

reneeyeow02 committed:
Level-7: Save

reneeyeow02 committed:
Level 8. Dates and Times

reneeyeow02 committed:
A-MoreOOP

reneeyeow02 committed:
A-Packages

reneeyeow02 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

reneeyeow02 committed:
A-JUnit

reneeyeow02 committed:
Gradle fixes

reneeyeow02 committed:
A-Jar

reneeyeow02 committed:
A-Jar

reneeyeow02 committed:
A-JavaDoc

reneeyeow02 committed:
A-CodingStandard

reneeyeow02 committed:
Level-9: Find

reneeyeow02 committed:
Change name from duke to meowmeow

jetlfj commented on src/main/java/meowmeow/Meowmeow.java:
```diff
@@ -0,0 +1,51 @@
+    private Ui ui;
+
+    /**
+     * Constructor for the Meowmeow class.
+     * @param filePath is a String that specifies the filepath of the save file.
```
Shouldn't there be an empty line between the description and parameter sections in Javadoc comments?
```suggestion
     * Constructor for the Meowmeow class.
     *
     * @param filePath is a String that specifies the filepath of the save file.
```

jetlfj commented on src/main/java/meowmeow/Parser.java:
```diff
@@ -0,0 +1,76 @@
+    public static Command parse(String userInput) {
+
+        try {
+            String[] splitUI = userInput.split(" ");
```
Should this variable be named differently to avoid confusion?

jetlfj commented on src/main/java/meowmeow/Parser.java:
```diff
@@ -0,0 +1,76 @@
+import meowmeow.commands.ListCommand;
+import meowmeow.commands.MarkCommand;
+
+public class Parser {
```
Would be good to have a header comment briefly describing what this class does.

jetlfj commented on src/main/java/meowmeow/TaskList.java:
```diff
@@ -0,0 +1,167 @@
+package meowmeow;
+import java.time.LocalDateTime;
```
Shouldn't there be an empty line between these two lines?
```suggestion
package meowmeow;

import java.time.LocalDateTime;
```

jetlfj commented on src/main/java/meowmeow/TaskList.java:
```diff
@@ -0,0 +1,167 @@
+        for (int i = 0; i < numOfInputs; i++) {
+            task = taskList.get(i);
+
+            if (task.isDone()) { //task is done
+                System.out.println((i + 1) + ". " + task);
+
+            } else { //task is not done
+                System.out.println((i + 1) + ". " + task);
+            }
```
Is there a need for this if-else statement?
```suggestion
            System.out.println((i + 1) + ". " + task);
```

jetlfj commented on src/main/java/meowmeow/MeowmeowException.java:
```diff
@@ -0,0 +1,29 @@
+ */
+public class MeowmeowException extends Exception {
+
+    private String message;
+
+    /**
+     * Constructor for MeowmeowException.
+     * @param message the message to be displayed to the user.
+     */
+    MeowmeowException(String message) {
+        this.message = message;
+    }
```
Is there a need for these when the superclass already has a message variable?
```suggestion
    /**
     * Constructor for MeowmeowException.
     * @param message The message to be displayed to the user.
     */
    MeowmeowException(String message) {
        super(message);
    }
```

reneeyeow02 replied:
I'm not sure, when I try to use the superclass message variable it causes issues with the toString method's return message so I didn't use the superclass variable.

jetlfj commented on src/main/java/meowmeow/MeowmeowException.java:
```diff
@@ -0,0 +1,29 @@
+        this.message = message;
+    }
+
+    /**
+     * Method that returns the message to be displayed to the user.
+     * @return the message to be displayed to the user.
+     */
+    @Override
+    public String toString() {
+        return message;
+    }
```
Similarly, the superclass' toString() method will return the message, so this is not necessary.

jetlfj reviewed (commented):
Overall, the code was easy to read. I noted a few minor coding standard violations, mainly the formatting of Javadoc comments throughout the code. The TaskList class is also getting quite long, consider putting the print statements in the UI class instead.

reneeyeow02 committed:
A-CheckStyle: Use CheckStyle

reneeyeow02 committed:
JavaFX

reneeyeow02 committed:
Level-10: GUI

reneeyeow02 committed:
A-Assertions

Use assert feature to document important assumptions in the code

Build new Jar file

reneeyeow02 committed:
A-CodeQuality

Clean up code, fixing indentation and removing redundant import statements

reneeyeow02 committed:
Merge pull request #2 from reneeyeow02/branch-A-Assertions

A-Assertions

reneeyeow02 committed:
Merge pull request #3 from reneeyeow02/branch-A-CodeQuality

A-CodeQuality

reneeyeow02 committed:
C-Undo

Add functionality to undo most recent command

reneeyeow02 committed:
Merge pull request #4 from reneeyeow02/branch-BCD-Extension

C-Undo

reneeyeow02 committed:
Create Ui.png

reneeyeow02 committed:
Code cleanup

reneeyeow02 committed:
A-UserGuide

Create User guide and fix bugs

reneeyeow02 committed:
Update .gitignore

reneeyeow02 committed:
Remove error logs

reneeyeow02 committed:
Merge branch 'master' of https://github.com/reneeyeow02/ip

reneeyeow02 committed:
Edit GUI

reneeyeow02 committed:
Update Ui.png

reneeyeow02 committed:
Add delay for exit command

Bot now prints a goodbye message before exiting the program

reneeyeow02 committed:
Update User guide

reneeyeow02 committed:
Handle more errors and improve GUI

Bot now gives feedback for more invalid user inputs

Improved GUI to allow resizing of program window

Hi message is now printed on program start

Disable send button when there is no input

reneeyeow02 committed:
Update Ui.png

reneeyeow02 committed:
Fix code style issues

quannam0124 commented on src/main/java/meowmeow/Storage.java:
```diff
@@ -0,0 +1,142 @@
+                    break;
+                }
+            }
+        } catch (FileNotFoundException e) {
```
This method is particularly long. Perhaps you could extract out to another method?

quannam0124 reviewed (commented):
Overall your code is not hard to read and looks good! Maybe you could take a look at particularly long methods to see if you could extract out to make it easier to read? Great job!

reneeyeow02 committed:
Handle IndexOutOfBounds error for mark and unmark

reneeyeow02 committed:
Fix and add Javadocs

reneeyeow02 committed:
Refactor code and update User Guide
